pylana.py

```diff
@@ -36,21 +41,31 @@ def uploadEventLog(self, logFile, logSemantics):
         return r
 
     def uploadEventLogWithCaseAttributes(self, logFile, logSemantics,
-                                         caseAttributeFile, caseAttributeSemantics):
-        endpoint = 'api/uploadEventLogWithCaseAttributes'
+                                         caseAttributeFile, caseAttributeSemantics, logName = None):
 
-        files = {
-            'eventCSVFile': open(logFile, 'rb'),
-            'caseAttributeFile': open(caseAttributeFile, 'rb'),
-        }
+        endpoint = 'api/logs/csv-case-attributes-event-semantics'
 
-        semantics = {
-            'eventSemantics': open(logSemantics).read(),
-            'caseSemantics': open(caseAttributeSemantics).read()
+        files = {
+            'eventCSVFile': (logFile.split('/')[-1], open(logFile, 'rb'), 'text/csv'),
+            'caseAttributeFile': (caseAttributeFile.split('/')[-1], open(caseAttributeFile, 'rb'), 'text/csv'),
         }
 
-        upload_response = requests.post(self.url + endpoint, headers=self.headers, files=files,
-                                        data=semantics, verify=False).json()
+        if not logName:
```

Comment:
Is the `request` library not able to handle optional values, meaning you simply put `None` as a value in the parameters and it won't send the dict entry?

Or will it send a stringified version of `None` ?

Reply:
Hm, doesn't look so good:
```python
>>> import requests
>>> response = requests.post('https://httpbin.org/post', json={'key':'value', 'key2': None})
>>> json_response = response.json()
>>> json_response
{u'files': {}, u'origin': u'35.207.142.139, 35.207.142.139', u'form': {}, u'url': u'https://httpbin.org/post', u'args': {}, u'headers': {u'Content-Length': u'30', u'Accept-Encoding': u'gzip, deflate', u'Accept': u'*/*', u'User-Agent': u'python-requests/2.22.0', u'Host': u'httpbin.org', u'Content-Type': u'application/json'}, u'json': {u'key2': None, u'key': u'value'}, u'data': u'{"key2": null, "key": "value"}'}
```

To me it seems like `None` is being sent as request data :(

Reply:
Actually I checked the documentary and it says 'Note that any dictionary key whose value is None will not be added to the URL's query string.' when using the `data` parameter. So

```python
>>> data = {"a": None, "b": "dummy request"}
>>> requests.Request('POST', 'localhost:4000', data = data).prepare().body
'b=dummy+request'
```

I will make another commit to simplify the code. Thank you for the suggestion!

Reply:
Done!

Reply:
Have you confirmed that this yields the expected result when run against LANA?
If you've tested it then it's good enough for me :)

Reply:
Tested against a sample data and it works perfectly!

```python
>>> res = mylana.uploadEventLogWithCaseAttributes('/Users/yunfanyang/Desktop/data/fixed/data-extractor/daimler-halle36/daimler-FWS-halle36.csv', '/Users/yunfanyang/Desktop/data/fixed/data-extractor/daimler-halle36/daimler-FWS-halle36-eventSemantics.txt', '/Users/yunfanyang/Desktop/data/fixed/data-extractor/daimler-halle36/daimler-FWS-halle36-caseAttributes.csv', '/Users/yunfanyang/Desktop/data/fixed/data-extractor/daimler-halle36/daimler-FWS-halle36-caseSemantics.txt')
>>> res.status_code
200
```